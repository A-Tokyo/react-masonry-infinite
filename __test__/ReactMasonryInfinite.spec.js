import React from 'react'
import renderer from 'react-test-renderer'
import ReactMasonryInfinite from '../src'

function createNodeMock(element) {
  if (element.type === 'div' && element.props.className === 'layout') {
    return document.createElement('div');
  }
  return null;
}

it('ReactMasonryInfinite correctly render', () => {
  const arrayForChildrens = [1, 2, 3, 4, 5, 6]
  const AdditionalSnapshot = renderer
    .create(
      (
        <ReactMasonryInfinite
          className='layout'
          loadMore={() => null}
        >
          {arrayForChildrens.map(i => (
            <div key={i.toString()}>
              {i}
            </div>
          ))}
        </ReactMasonryInfinite>
      ),
      { createNodeMock }
    )
    .toJSON()
  expect(AdditionalSnapshot).toMatchSnapshot()
})
