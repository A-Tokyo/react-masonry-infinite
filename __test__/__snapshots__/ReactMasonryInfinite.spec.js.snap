// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactMasonryInfinite correctly render 1`] = `
<div>
  <div
    className="layout"
    style={Object {}}
  >
    <div>
      1
    </div>
    <div>
      2
    </div>
    <div>
      3
    </div>
    <div>
      4
    </div>
    <div>
      5
    </div>
    <div>
      6
    </div>
  </div>
</div>
`;
